### PR TITLE
Use string match to classify DNS error.

### PR DESCRIPTION
**Pull request by tcnghia**
It seems that typed-check aren't robust enough, so we are switching to
checking the error message here.

### CODE REVIEW

File: test/spoof/error_checks.go [hunk 1]
```diff
@@ -30,14 +29,13 @@ func isTCPTimeout(e error) bool {
...
-			}
-		}
+	if err == nil {
+		return false
 	}
-	return false
+	// Checking by casting to url.Error and casting the nested error
+	// seems to be not as robust as string check.
+	msg := strings.ToLower(err.Error())
+	return strings.Contains(msg, "no such host") || strings.Contains(msg, ":53")
```
Comment (srinivashegde86) [inline test/spoof/error_checks.go]: can we add a sample err string in the comment? I would like to understand what :53 means here

Comment (tcnghia) [reply test/spoof/error_checks.go]: Done.  :53 is the default port for DNS

Comment (srinivashegde86) [reply test/spoof/error_checks.go]: thanks :+1:

**Pull request merged**